### ISSUE TITLE
Get SQLDialect.VERTICA up to date

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14739]
Continuing after #11650 and #11681, A variety of recently added features / improvements have not yet been applied to the `VERTICA` dialect. These include:

- [ ] #11653 Array support
- [ ] #11657 Native `BITCOUNT` support

Without separate issue:

- [ ] `LISTAGG` support
- [ ] `MEDIAN` and `PERCENTILE_CONT`, `PERCENTILE_DISC` support
